lib/cli-engine.js

```diff
@@ -162,14 +165,33 @@ function processFile(filename, configHelper) {
         config,
         text,
         messages,
-        stats;
+        stats,
+        fileExtension = path.extname(filename),
+        processor;
 
     if (fs.existsSync(filePath)) {
         debug("Linting " + filePath);
         config = configHelper.getConfig(filePath);
         loadPlugins(config.plugins);
         text = fs.readFileSync(path.resolve(filename), "utf8");
-        messages = eslint.verify(text, config, filename);
+
+        for (var plugin in loadedPlugins) {
+            if (loadedPlugins[plugin].processors && Object.keys(loadedPlugins[plugin].processors).indexOf(fileExtension) >= 0) {
+                processor = loadedPlugins[plugin].processors[fileExtension];
+                break;
+            }
+        }
+
+        if (processor) {
+            var parsedBlocks = processor.preprocess(text);
```

Comment:
Hmmm.. That's an interesting question. I'm not sure. I can't come up with the situation where you would need one but not the other. In my mind, if you need to do something to text of the file before validating it, you also always have to do something with the output (correct locations). And visa-versa.